### ISSUE TITLE
Data too long for column error

### ISSUE THREAD

**Issue by keyurshah** [barrelstrength/craft-sprout-email#167]
### Description (screenshots, logs, ...)
Hello,

Thanks for this wonderful plugin. I'm not seeing some emails show up in the Sprout log, but I do see them go out in the mailgun logs.

I scanned the logs, and am seeing this error

```
2020-07-20 08:18:21 [-][150629][-][error][barrelstrength\sproutbasesentemail\services\SentEmails::saveSentEmail] SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'body' at row 1
The SQL being executed was: INSERT INTO `sproutemail_sentemail` (`id`, `title`, `emailSubject`, `fromEmail`, `fromName`, `toEmail`, `body`, `htmlBody`, `info`, `status`, `dateCreated`, `dateUpdated`, `uid`) VALUES 
```

Can the body column be adjusted to accommodate the data?

### Steps to reproduce

1. Send an email with longer text in the body

### Additional info

- Craft version: 3.4.29.1
- PHP version: 7.4
- Database driver & version:  
- Plugins & versions:  Sprout 1.1.1

Thanks!

**Comment by BenParizek** [comment]
Hey @keyurshah 

We did increase the size of the `htmlBody` column in [Sprout Email v4.4.4](https://github.com/barrelstrength/craft-sprout-email/blob/v4/CHANGELOG.md#444---2020-05-21) and [Sprout Sent Email v1.1.1](https://github.com/barrelstrength/craft-sprout-sent-email/blob/v1/CHANGELOG.md#111---2020-05-21)

However, it looks like you're running into the issue with the `body` column. We'll add another migration for that column although it's surprising to me that the text version of the email needs to be that long. Typically the longer length is due to the mess of HTML required to format a long HTML email, not due to the content of the email. So you may also want to take a look at what's getting sent out as the text version for your email.

In any case, until we get that migration out, you can modify your database directly and update the `sproutemail_sentemail.body` column to use the data type MEDIUM_TEXT and that should solve the issue and not be harmed when we do add the migration.

**Comment by engram-design** [comment]
Running into this issue too. I would submit a PR, but I can't figure out which repository to submit it to (base-email, etc). Notably, the content is HTML content, so maybe that's something on our end, where we should be providing a plain-text version of the email?

```
SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'body' at row 1
The SQL being executed was: INSERT INTO `sproutemail_sentemail` (`id`, `title`, `emailSubject`, `fromEmail`, `fromName`, `toEmail`, `body`, `htmlBody`, `info`, `status`, `dateCreated`, `dateUpdated`, `uid`) VALUES (151043, 'Your order is confirmed ', 'Your order is confirmed ', 'sales@site.com.au', 'Site', 'customer@site.com.au', '<!doctype html>\n<html xmlns=\"http://www.w3.org/1999/xhtml\" xmlns:v=\"urn:schemas-microsoft-com:vml\" xmlns:o=\"urn:schemas-microsoft-com:office:office\">\n\n<head>\n  <title> Your order is confirmed </title>\n  <!--[if !mso]><!-- -->\n  <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n  <!--<![endif]-->\n  <meta http-equiv=\"Content-Type\" ...
```

**Comment by BenParizek** [comment]
@engram-design The column size will be addressed in a migration in our next major release. Can you modify the `sproutemail_sentemail.body` column to use the data type MEDIUM_TEXT directly for now?

Based on your comment that the content is HTML, I will look into that. Each email can have an `email.html` and `email.txt` file, so perhaps there is a bug around something there. Can you share any details about how you have things setup around HTML and text versions of the email?

**Comment by engram-design** [comment]
Yep, just doing a manual column update for the moment, otherwise our Sentry logs are on fire!

I will flag that we're only using this plugin on this site to track sent emails (before your sent emails plugin existed), so probably something around that. No drama though.
